python/pyarrow/tests/test_compute.py

```diff
@@ -1255,6 +1255,78 @@ def test_strptime():
     assert got == expected
 
 
+def _check_datetime_components(timestamps, timezone=None):
+    import pandas as pd
+
+    if timezone:
+        ts = pd.to_datetime(timestamps).tz_localize(timezone).to_series()
+    else:
+        ts = pd.to_datetime(timestamps).to_series()
+
+    tsa = pa.array(ts)
+
+    subseconds = ((ts.dt.microsecond * 10**3 +
+                  ts.dt.nanosecond) * 10**-9).round(9)
+    iso_calendar_fields = [
+        pa.field('iso_year', pa.int64()),
+        pa.field('iso_week', pa.int64()),
+        pa.field('iso_day_of_week', pa.int64())
+    ]
+
+    iso_year = ts.dt.isocalendar()["year"].astype(int)
+    iso_week = ts.dt.isocalendar()["week"].astype(int)
+    iso_day = ts.dt.isocalendar()["day"].astype(int)
+    iso_calendar = pa.StructArray.from_arrays(
+        [iso_year, iso_week, iso_day], fields=iso_calendar_fields)
+
+    assert pc.year(tsa).equals(pa.array(ts.dt.year))
+    assert pc.month(tsa).equals(pa.array(ts.dt.month))
+    assert pc.day(tsa).equals(pa.array(ts.dt.day))
+    assert pc.day_of_week(tsa).equals(pa.array(ts.dt.day_of_week))
+    assert pc.day_of_year(tsa).equals(pa.array(ts.dt.day_of_year))
+    assert pc.iso_year(tsa).equals(pa.array(iso_year))
+    assert pc.iso_week(tsa).equals(pa.array(iso_week))
+    assert pc.iso_calendar(tsa).equals(iso_calendar)
+    assert pc.quarter(tsa).equals(pa.array(ts.dt.quarter))
+    assert pc.hour(tsa).equals(pa.array(ts.dt.hour))
+    assert pc.minute(tsa).equals(pa.array(ts.dt.minute))
+    assert pc.second(tsa).equals(pa.array(ts.dt.second.values))
+    assert pc.millisecond(tsa).equals(pa.array(ts.dt.microsecond // 10**3))
+    assert pc.microsecond(tsa).equals(pa.array(ts.dt.microsecond % 10**3))
+    assert pc.nanosecond(tsa).equals(pa.array(ts.dt.nanosecond))
+    assert pc.subsecond(tsa).equals(pa.array(subseconds))
+
+
+@pytest.mark.pandas
+def test_extract_datetime_components():
+    # TODO: see https://github.com/pandas-dev/pandas/issues/41834
+    # "1899-01-01T00:59:20.001001001"
```

Comment:
Probably best to remove this since I believe pandas is wrong and it's not really our TODO.
It's kinda hard to confirm who is right.

Reply:
Removed.